featuremanagement/_models/_feature_flag.py

```diff
@@ -115,9 +115,9 @@ def variants(self):
 
     def _validate(self):
         if not isinstance(self._id, str):
-            raise ValueError("Feature flag id field must be a string.")
+            raise ValueError(f"Invalid setting 'id' with value '{self._id}' for feature '{self._id}'.")
```

Comment:
If the class of this instance does not have a __str__ or __repr__ defined, the value could appear weird in the error message. we say something along these lines:
 `Invalid type for setting 'id'. 'str'  expected but {type(self._id).__name__} given`

Reply:
This change was to start moving us towards consistent error messages between the libraries, this is the exact way DotNet does the message.

Reply:
I like the idea of being consistent, but in dotnet- I don't think this error will ever happen on name/id and I think the wording is weird here for the str check.

This error will only happen if i'ts not a string, but when logging it would say something like:
`Invalid setting 'id' with value 'null' for feature 'null'` - which is kind of clear, but number and object could be weird: `Invalid setting 'id' with value '1' for feature '1'`, because string:"1" is a valid name but number:1 is not. 

I prefer something closer to @albertofori 's suggestion.

Reply:
This isn't a great example as 1 is valid as python is typeless.

I can update this in a sepreate PR, I can change to what Albert mentioned, it's the change bellow this one that is 1 to 1 with Dot Net.